src/compiler/scanner.rs

```diff
@@ -78,21 +81,21 @@ impl Scanner {
             .intersects(TokenFlags::PrecedingLineBreak)
     }
 
-    fn get_identifier_token(&mut self) -> SyntaxKind {
+    fn get_identifier_token(&self) -> SyntaxKind {
         let len = self.token_value().len();
         if len >= 2 && len <= 12 {
             let ch = self.token_value().chars().nth(0).unwrap();
             if ch >= CharacterCodes::a && ch <= CharacterCodes::z {
-                let keyword = text_to_keyword.get(self.token_value());
+                let keyword = text_to_keyword.get(&self.token_value());
```

Comment:
This is what caused me to change `text_to_keyword` from `HashMap<&'static str, SyntaxKind>` to `HashMap<String, SyntaxKind>`, for some reason when I changed `self.token_value()` to return a `String` rather than an `&str` (because now that `token_value` is wrapped in an `RwLock` it doesn't want to return an `&str` reference to its contents) it was unhappy with me passing `&self.token_value()` as an argument to `text_to_keyword.get()`